pkg/generate/git/repository.go

```diff
@@ -337,6 +348,42 @@ func command(w io.Writer, name, dir string, env []string, args ...string) (stdou
 	return
 }
 
+// timedCommand executes an external command in the given directory with a timeout.
+// The command's standard out and error are trimmed and returned as strings
+// It may return the type *GitError if the command itself fails.
+func timedCommand(timeout time.Duration, w io.Writer, name, dir string, env []string, args ...string) (stdout, stderr string, err error) {
+	// typedWrapper wraps `command()` and returns a single, typed value in order for us to pipe it into a channel
+	typedWrapper := func(w io.Writer, name, dir string, env []string, args ...string) cmdResult {
```

Comment:
I don't want to mess with the signature of the existing function, as this container type isn't meaningful for anyone but this channel method.